Jsonp

bengillies opened the pull request:
This is for the jsonp stuff.

To keep everything safe, I've added some extra code to DropPrivs to drop privileges when the user specifies a jsonp_callback query parameter. This should ensure that only public stuff is ever sent as jsonp...

colmjude committed:
These changes set the default privacy mode to public.

There is now an option in tweaks that, when checked, sets the privacy mode to private.
It is initially set in TiddlySpaceConfig.js so that it still works even if System-Plugins is unincluded

colmjude committed:
Defines config.optionsDesc in fixtures.js

Required for an exception not to be thrown when running tests given changes for public by default

Commit:
modified configuration to use latest TiddlyWiki beta

NB: This doesn't actually work yet, as apparently TiddlyWebWiki's
setting takes precedence. Needs investigation.

Commit:
ensured base_tiddlywiki takes precedence over tiddlywebwiki's setting

more instancer weirdness...

Commit:
replaced default use of TiddlyWiki beta with an optional serializer

accessed via bwiki extension or text/x-btiddlywiki content type

The current TiddlyWiki beta release is not yet stable enough to be used
by default, so we're offering it as an option instead:
http://tiddlyspace.tiddlyspace.com/Use%20TiddlyWiki%20Nightly%2FBeta%20Releases

Commit:
ensured cloneCommand does not override inherited fields

if a new tiddler inherits fields (usually via a tiddlylink), we want to
use the regular editTiddler command instead of cloneTiddler, ensuring
that relationship takes precedence

(commit 521f29947697d69a6d668791d6a97b462c415341 explains why
cloneTiddler is used for non-existing tiddlers in the first place)

fixes #482

Commit:
modified beta serializer to be activated via URL parameter

This is more flexible than using a separate content type, e.g. allowing
it to be used with the root handler.

Note that this replaces TiddlyWebWiki's default serializer with a
subclass of tiddlywebwiki.serialization.

Commit:
Merge branch 'tiddlywiki_nightly'

Commit:
release 0.9.45

* added upload macro to backstage (Space > Options)
* added option to use TiddlyWiki beta release (via "twrelease=beta" URL
  parameter)
* updated default SiteIcon
* fixed issue regarding inherited fields

Commit:
post-merge cleanup

* fixed typos ("priavte" - indicates that tests are required!)
* fixed STYLE issues
* updated use of String's format method
* removed unnecessary shortcut variables (they're particularly
  unsuitable in TiddlySpaceConfig, though that's mostly an indication
  that plugin has become too long and messy and needs refactoring)

Commit:
Merge remote branch 'colmjude/PrivacyMode'

bengillies committed:
Added support for jsonp

TiddlySpace now requires tiddlywebplugins.jsonp.

Also added some code to DropPrivs that drops privileges to GUEST
if the user is trying to use jsonp, i.e. if there is a jsonp_callback
parameter in tiddlyweb.query.

Also fixed some indentation

bengillies committed:
added some tests to ensure that private data is safe

Commit:
Merge remote branch 'jdlrobson/BinaryErrors'

Commit:
Merge remote branch 'bengillies/jsonp' into jsonp

Commit:
added additional JSONP tests, highlighting a possible vulnerability

intentionally failing test:
private recipe entities can be accessed via JSONP

plus PEP8-related corrections and stylistic adjustments

FND commented:
I found a potential vulnerability: FND@33c000174f90fa5f126de2bf3ac6ad9a8bd296ab

Let's discuss on IRC.

bengillies committed:
Added policies to make_fake_space and updated tests to expect this

Some of the tests in test_jsonp.py require the policy set properly on the bags/
recipes. The most obvious way of doing thhis was to update make_fake_space to
set the policies of each bag/recipe correctly. Doing this broke some other tests
though (ie tests that didn't expect a policy on the private bag), so I've also
updated them to make their requests as a logged in user as well.

bengillies committed:
added some tests to test the root domain as well as subdomains

As it's jsonp, we need to check the root domain as well as subdomains.
Previously, this was safe and being ignored, but with jsonp, its no longer
safe, so we need to DropPrivs for everything whenever jsonp_callback
is specified.

This resulted in a small bit of refactoring on DropPrivs and the introduction
of a new method "_handle_jsonp" which _always_ gets called. This should result
in the desired behaviour (dropping privs) whatever url is called.
